### DISCUSSION TITLE
Custom error message for z.literal()

### DISCUSSION THREAD

**Discussion by MerlinMason** [colinhacks/zod#1525] [Q&A] [answered]
I'm using zod with react-hook-form to validate form fields. There's a checkbox to accept legal terms and I want to validate it's checked and if not show a message.

I've tried passing `invalid_type_error`, `required_error` and `message` into the options object for `literal` but none seem to work. 

```ts
const SignUpSchema = z.object({
    // this works fine
    password: z.string().min(10, { message: "Password must be at least 10 characters" }), 

    // displays: "Invalid literal value, expected true" 😭 
    legal: z.literal(true, { required_error: "You must accept the terms & conditions" }), 
});
```

Is this possible? 
If not, how else would you go about achieving a similar result?

**Comment by MerlinMason** [comment, marked as answer]
Allow me to answer my own question 🎉 

```ts
legal: z.literal(true, {
    errorMap: () => ({ message: "You must accept the terms & conditions" }),
}),
```

The inconsistencies of displaying error messages is quite confusing :(

**Reply by CanRau** [reply]
indeed very confusing, was trying `invalid_type_error` as well as `required_error` which both exist in `literal` options according to TypeScript but don't seem to do anything 🥺

Thanks for sharing your solution/workaround Merlin 🙏🏼

**Reply by laneme** [reply]
When using with the `zodResolver` from `react-hook-form`, my original errormap (passed via zodResolver) was getting overriden entirely by this errormap.

Not sure where the issue is but refinements worked for me

```ts
z.coerce.boolean().refine(bool => bool == true, {
    message: 'You must agree to our terms and conditions'
})
```
